includes/class-edd-fees.php

```diff
@@ -196,6 +199,22 @@ public function get_fees( $type = 'fee', $download_id = 0 ) {
 			}
 
 		}
+		
+		//Now that we've removed any fees that are for other Downloads, lets also remove any fees that don't match this price id (or Variable Price)
+		if( ! empty( $fees ) && ! empty( $download_id ) && ! empty( $price_id ) ) {
```

Comment:
For price ID, we'll need to explicitly check that it is an `int` that is `0` or greater. Passing in `0`, which is a valid ID for some cases, would not pass the `! empty()` check.